Sources/StreamChatUI/ChatMessageList/RETHINK/MessageListVC.swift

```diff
@@ -118,4 +232,181 @@ class MessageListVC<ExtraData: ExtraDataTypes>: _ViewController, UICollectionVie
         
         return cell
     }
+    
+    // TODO: Remove when proper handler is implemented
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        didSelectMessageCell(at: indexPath)
+    }
+    
+    /// Will scroll to most recent message on next `updateMessages` call
+    public func setNeedsScrollToMostRecentMessage(animated: Bool = true) {
+        needsToScrollToMostRecentMessage = true
+        needsToScrollToMostRecentMessageAnimated = animated
+    }
+
+    /// Force scroll to most recent message check without waiting for `updateMessages`
+    public func scrollToMostRecentMessageIfNeeded() {
+        guard needsToScrollToMostRecentMessage else { return }
+        
+        scrollToMostRecentMessage(animated: needsToScrollToMostRecentMessageAnimated)
+    }
+
+    public func scrollToMostRecentMessage(animated: Bool = true) {
+        needsToScrollToMostRecentMessage = false
+
+        // our collection is flipped, so (0; 0) item is most recent one
+        collectionView.scrollToItem(at: IndexPath(item: 0, section: 0), at: .bottom, animated: animated)
+    }
+
+    public func updateMessages(with changes: [ListChange<_ChatMessage<ExtraData>>], completion: ((Bool) -> Void)? = nil) {
+        collectionView.performBatchUpdates {
+            for change in changes {
+                switch change {
+                case let .insert(_, index):
+                    collectionView.insertItems(at: [index])
+                case let .move(_, fromIndex, toIndex):
+                    collectionView.moveItem(at: fromIndex, to: toIndex)
+                case let .remove(_, index):
+                    collectionView.deleteItems(at: [index])
+                case let .update(_, index):
+                    collectionView.reloadItems(at: [index])
+                }
+            }
+        } completion: { flag in
+            completion?(flag)
+            self.scrollToMostRecentMessageIfNeeded()
```

Comment:
This doesn't seem right. Why we should scroll to the last message when anything changes in the messages?

Reply:
taken from current implementation in `ChatMessageListVC`